winit-wayland/src/window/mod.rs

```diff
@@ -103,17 +111,91 @@ impl Window {
             WindowDecorations::RequestClient
         };
 
-        let window =
-            state.xdg_shell.create_window(surface.clone(), default_decorations, &queue_handle);
-
-        let mut window_state = WindowState::new(
-            event_loop_window_target.handle.clone(),
-            &event_loop_window_target.queue_handle,
-            &state,
-            size,
-            window.clone(),
-            attributes.preferred_theme,
-        );
+        let (window, mut window_state) = if wl_attrs
+            .as_ref()
+            .is_some_and(|wl_attrs| wl_attrs.is_layer_shell)
+            || attributes.window_level != WindowLevel::Normal
```

Comment:
I think the layer shell should be used only on purpose, it should not be related to the window level. e.g. I would not expect an error on creation of window on Gnome just because I wanted always on top window

Reply:
First of all, thanks for your contribution, the problem with this is that xdg doesn't support levels anyway, so you won't get the expected behavior in wayland, this is only for layer_shell's

Reply:
Currently that is true, but as wayland gets more mature that may change